src/blog/2024/04/using-webhook-with-node-red.md

```diff
@@ -0,0 +1,188 @@
+- Data Reception by Third-Party Application: The third-party application will receive the data sent via the WebHook to the designated URL or listener provided during registration.
+- Custom Action Execution: Upon receiving the POST request, specific actions can be performed.
+
+## API Vs Webhook
+
+It's common and understandable to get confused between APIs and webhooks, especially when you are learning about webhooks for the first time. However, comparing the two can help dispel these confusions.
+
+## API:
+
+- **Direction:**
+  - Typically involves client-to-server communication.
+- **Initiation:**
+  - Requests are initiated by the client.
+- **Request Method:**
+  - Usually employs HTTP methods like GET, POST, etc.
+- **Response:**
+  - Provides an immediate response upon request.
+- **Data Transfer:**
+  - Utilizes a pull model where the client fetches data.
+- **Polling:**
+  - Requires periodic polling for updates.
+- **Payload:**
+  - The client specifies the payload in the request.
+- **Error Handling:**
+  - Typically includes error codes and messages.
+
+## Webhook:
+
+- **Direction:**
+  - Typically involves server-to-server communication.
+- **Initiation:**
+  - Requests are initiated by the server.
+- **Request Method:**
+  - Typically uses the HTTP POST method.
+- **Response:**
+  - Does not provide an immediate response; asynchronous.
+- **Data Transfer:**
+  - Operates on a push model where the server pushes data to the client.
+- **Polling:**
+  - No need for polling; receives updates directly.
+- **Payload:**
+  - The server defines the payload in the outgoing request.
+- **Error Handling:**
+  - Errors are handled by retry mechanisms or manual intervention.
+
+### Example Scenario:
+
+For example, Consider a manufacturing facility that utilizes temperature sensors to monitor temperature levels in critical areas. When the temperature surpasses predefined thresholds, it triggers a series of actions for maintenance and monitoring.
+
+!["Diagram explaining how component works in"](./images/using-webhook-with-node-red-diagram.png "Role Based Access Control For Node-RED with FlowFuse"){data-zoomable}
+
+- Temperature Sensor (Server 1): These physical sensors are installed in the manufacturing plant to monitor temperature. They trigger webhook requests to Server 2 whenever abnormal temperature patterns are detected.
+
+- Server 2 (Webhook Server): This server creates and hosts the webhook endpoint. It receives HTTP requests from the temperature sensors (Server 1) when abnormal temperatures are detected. The request contains temperature data. Server 2 then processes this data and sends a POST request with relevant information to Server 3.
```

Comment:
Why have this intermediary server? Does this server process a lot of sensor readings?

Server 1 has limitations on how many messages it will send, so I believe you should explain why server 1 doesn't send the message straight to server 3.

Reply:
Okay ZJ, I'll provide an explanation for why Server 1 doesn't directly send messages to Server 3.